File: maximum-subarray.cc
class Solution {
public:
    int maxSubArray(int A[], int n) {
        int ans = INT_MIN, sum = 0;
        for (int i = 0; i < n; i++) {
            sum += A[i];
            if (sum > ans) ans = sum;
            if (sum < 0) sum = 0;
        }
        return ans;
    }
};

File: maximum-subarray.java
public class Solution {
    public int maxSubArray(int[] A) {
        int ans = Integer.MIN_VALUE;
        int val = 0;
        for (int elem : A) {
            val += elem;
            if (val > ans) ans = val;
            if (val < 0) val = 0;
        }
        return ans;
    }
}
